charms/katib-ui/src/charm.py

```diff
@@ -29,89 +30,75 @@ def __init__(self, msg, status_type=None):
-                            "rules": [
-                                {
-                                    "apiGroups": [""],
-                                    "resources": ["configmaps", "namespaces"],
-                                    "verbs": ["*"],
-                                },
-                                {
-                                    "apiGroups": ["kubeflow.org"],
-                                    "resources": [
-                                        "experiments",
-                                        "trials",
-                                        "suggestions",
-                                    ],
-                                    "verbs": ["*"],
-                                },
-                            ],
-                        }
-                    ]
-                },
-                "containers": [
-                    {
-                        "name": "katib-ui",
-                        "command": ["./katib-ui"],
-                        "args": [f"--port={self.model.config['port']}"],
-                        "imageDetails": image_details,
-                        "ports": [
-                            {"name": "http", "containerPort": self.model.config["port"]}
-                        ],
-                        "envConfig": {"KATIB_CORE_NAMESPACE": self.model.name},
-                    }
-                ],
-            }
-        )
-
-        self.model.unit.status = ActiveStatus()
+    @property
+    def container(self):
+        return self._container
+
+    @property
+    def _katib_ui_layer(self) -> Layer:
+        layer_config = {
+            "summary": "katib-ui-operator layer",
+            "description": "pebble config layer for katib-ui-operator",
+            "services": {
+                self._container_name: {
+                    "override": "replace",
+                    "summary": "entrypoint of the katib-ui-operator image",
+                    "command": f"./katib-ui --port={self._port}",
+                    "startup": "enabled",
+                    "environment": {"KATIB_CORE_NAMESPACE": self.model.name},
+                }
+            },
+        }
+        return Layer(layer_config)
+
+    def _update_layer(self) -> None:
+        """Updates the Pebble configuration layer if changed."""
+        current_layer = self.container.get_plan()
+        new_layer = self._katib_ui_layer
+        self.logger.debug(f"NEW LAYER: {new_layer}")
+        if current_layer.services != new_layer.services:
+            self.unit.status = MaintenanceStatus("Applying new pebble layer")
+            self.container.add_layer(self._container_name, new_layer, combine=True)
+            try:
+                self.logger.info(
+                    "Pebble plan updated with new configuration, replanning"
+                )
+                self.container.replan()
+            except ChangeError:
+                raise CheckFailed("Failed to replan", BlockedStatus)
```

Comment:
You can actually just use the update [layer function](https://github.com/canonical/charmed-kubeflow-chisme/blob/main/src/charmed_kubeflow_chisme/pebble/README.md#update_layer) from chisme.